pyteal/ast/opup.py

```diff
@@ -87,35 +87,43 @@ def __init__(self, mode: OpUpMode, target_app_id: Expr = None):
 
         self.mode = mode
 
-    def _construct_itxn(self) -> Expr:
+    def _construct_itxn(self, inner_fee: Expr = None) -> Expr:
+        if inner_fee is not None:
+            require_type(inner_fee, TealType.uint64)
+        else:
+            inner_fee = MIN_TXN_FEE
+
         if self.mode == OpUpMode.Explicit:
-            return Seq(
-                InnerTxnBuilder.Begin(),
-                InnerTxnBuilder.SetFields(
-                    {
-                        TxnField.type_enum: TxnType.ApplicationCall,
-                        TxnField.application_id: self.target_app_id,
-                    }
-                ),
-                InnerTxnBuilder.Submit(),
+            return InnerTxnBuilder.Execute(
+                {
+                    TxnField.type_enum: TxnType.ApplicationCall,
+                    TxnField.application_id: self.target_app_id,
+                    TxnField.fee: inner_fee,
```

Comment:
The idea is that the inner txn fee should specifically _not_ be set if `inner_fee` is None.

Reply:
ah ok, will change